viewing_party/party.py

```diff
@@ -0,0 +1,150 @@
+    for friends in amandas_data["friends"]:
+        for movie in friends["watched"]:
+            if movie not in friends_watched:
+                friends_watched.append(movie)
+
+    friends_unique_watched = []
+    for movie in friends_watched:
+        if movie not in amandas_data["watched"]:
+            friends_unique_watched.append(movie)
+    
+    return friends_unique_watched
+
+def get_available_recs(amandas_data):
+    # creates a list of movies in user's friends watched lists
+    friends_watched_mov = []
+    for friends in amandas_data["friends"]:
+        for movie in friends["watched"]:
+            if movie not in friends_watched_mov:
+                friends_watched_mov.append(movie)
+    
+    # creates a list of the services user's friends have
+    friends_watched_serv = []
+    for friends in amandas_data["friends"]:
+        for movie in friends["watched"]:
+            if movie["host"] not in friends_watched_serv:
+                friends_watched_serv.append(movie["host"])
+    
+    # adds movies to available recommendation list if user is subscribed to that service
+    available_recs = []
+    for service in friends_watched_serv:
+        if service in amandas_data["subscriptions"]:
+            for item in friends_watched_mov:
+                if service == item["host"]:
+                    available_recs.append(item)
+    
+    # removes movies from available recommendations if user has already watched them
+    for movie in amandas_data["watched"]:
+        for rec in available_recs:
+            if movie["title"] in rec.values():
+                available_recs.remove(rec)
+
+    return available_recs
+
+def get_new_rec_by_genre(sonyas_data):
+    # consider making this a helper function - it's used 3 times accross difference functions
+    friends_watched = []
+    for friend in sonyas_data["friends"]:
+        for movie in friend["watched"]:
+            if movie not in friends_watched:
+                friends_watched.append(movie)
+    
+    sonyas_genres = []
+    for movie in sonyas_data["watched"]:
+        if movie["genre"] not in sonyas_genres:
+            sonyas_genres.append(movie["genre"])
+    
+    recommendations = []
+    for movie in friends_watched:
+        if movie["genre"] in sonyas_genres and movie not in sonyas_data["watched"]:
+            recommendations.append(movie)
+    
+    return recommendations
+
+def get_rec_from_favorites(sonyas_data):
+    favorites = []
+    for movie in sonyas_data["favorites"]:
+        if movie not in favorites:
+            favorites.append(movie)
+
+    friends_watched = []
+    for friend in sonyas_data["friends"]:
+        for movie in friend["watched"]:
+            if movie not in friends_watched:
+                friends_watched.append(movie)
+    
+    # remove movie from favorites if a friend has already seen it
+    for movie in friends_watched:
+        if movie in favorites:
+            favorites.remove(movie)
+    
+    return favorites
+
+
```

Comment:
Quick-tip: If you want to make your project neater and follow pep8 guidelines, you can have VS Code autoformat your files by right-clicking on a file in VS Code and select Format Document With... then select python from the drop-down menu. You'll then be prompted to install autopep8, click yes and the package will be installed within your virtual environment.